fe/fe-core/src/main/java/com/starrocks/catalog/LocalTablet.java

```diff
@@ -0,0 +1,609 @@
+            Replica replica = iterator.next();
+            if (replica.getBackendId() == backendId) {
+                iterator.remove();
+                Catalog.getCurrentInvertedIndex().deleteReplica(id, backendId);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Deprecated
+    public Replica deleteReplicaById(long replicaId) {
+        Iterator<Replica> iterator = replicas.iterator();
+        while (iterator.hasNext()) {
+            Replica replica = iterator.next();
+            if (replica.getId() == replicaId) {
+                LOG.info("delete replica[" + replica.getId() + "]");
+                iterator.remove();
+                return replica;
+            }
+        }
+        return null;
+    }
+
+    // for test,
+    // and for some replay cases
+    public void clearReplica() {
+        this.replicas.clear();
+    }
+
+    @Override
+    public void write(DataOutput out) throws IOException {
+        super.write(out);
+
+        out.writeLong(id);
+        int replicaCount = replicas.size();
+        out.writeInt(replicaCount);
+        for (int i = 0; i < replicaCount; ++i) {
+            replicas.get(i).write(out);
+        }
+
+        out.writeLong(checkedVersion);
+        out.writeLong(0); // write a version_hash for compatibility
+        out.writeBoolean(isConsistent);
+    }
+
+    @Override
+    public void readFields(DataInput in) throws IOException {
+        super.readFields(in);
+
+        id = in.readLong();
+        int replicaCount = in.readInt();
+        for (int i = 0; i < replicaCount; ++i) {
+            Replica replica = Replica.read(in);
+            if (deleteRedundantReplica(replica.getBackendId(), replica.getVersion())) {
+                replicas.add(replica);
+            }
+        }
+
+        if (Catalog.getCurrentCatalogJournalVersion() >= 6) {
+            checkedVersion = in.readLong();
+            in.readLong(); // read a version_hash for compatibility
+            isConsistent = in.readBoolean();
+        }
+    }
+
+    public static LocalTablet read(DataInput in) throws IOException {
+        LocalTablet tablet = new LocalTablet();
+        tablet.readFields(in);
+        return tablet;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (!(obj instanceof LocalTablet)) {
+            return false;
+        }
+
+        LocalTablet tablet = (LocalTablet) obj;
+
+        if (replicas != tablet.replicas) {
+            if (replicas.size() != tablet.replicas.size()) {
+                return false;
+            }
+            int size = replicas.size();
+            for (int i = 0; i < size; i++) {
+                if (!tablet.replicas.contains(replicas.get(i))) {
+                    return false;
+                }
+            }
+        }
+        return id == tablet.id;
+    }
+
+    @Override
+    public long getDataSize() {
```

Comment:
is it logical right to return the total size of each replica? and it seems a little different from the original Tablet::getDataSize(boolean singleReplica), may be you should check this carefully

Reply:
There is only one place `MaterializedIndex.getDataSize()` where this parameter is used, and it is false.

Reply:
LGTM